### ISSUE TITLE
Need documentation for pydoop MapReduce

### ISSUE THREAD

**Issue by vkuznet** [crs4/pydoop#174]
Hi, I'm trying to understand the example given on http://crs4.github.io/pydoop/examples/avro.html but I barely follow. In particular, I don't understand the meaning of parameters given to pydoop submit script, e.g. this command
```
export STATS_SCHEMA=$(cat stats.avsc)
pydoop submit \
  -D pydoop.mapreduce.avro.value.output.schema="${STATS_SCHEMA}" \
  --avro-input v --avro-output v \
  --upload-file-to-cache color_count.py --mrv2 \
  color_count input output
```
does not say what parameters means What is ```v``` for avro-input/avro-output options, what is ```color_count``` (I presume it's a name of python script), what is ```input``` and ```output```?

It would be nice to have proper documentation about these details. Also, I'm interested to see full example how to submit MapReduce job which will read data from HDFS and do its job. I don't see where HDFS is configured and how it is passed to the job. I don't know how to specify location of my data on HDFS for MapReduce job, etc. Please help.
Thanks,
Valentin.

**Comment by ilveroluca** [comment]
Hi Valentin,

yeah, no one's made the time yet to write detailed docs for the more recent features; at the moment the best place to look may be the `--help` message.  The `--avro-input` and `--avro-output` arguments can take the values `k`, `v`, or `kv` for key, value, or key and value, respectively.  This determines whether or not the data coming in on the relevant key/value input/output is interpreted as avro.

The mandatory arguments (`color_count`, `input`, `output`) are the regular positional arguments for the `submit` command:

```
[pireddu@pear pydoop] pydoop submit --help
...<snip>

positional arguments:
  MODULE            The module containing the Python MapReduce program
  INPUT                 input path to the maps
  OUTPUT             output path from the reduces
...
```

The default file system, which I think is what you're referring to in your last comments, is detected from the Hadoop configuration that is picked up by Pydoop -- like all hadoop commands.  You can specify the configuration to be used by setting the `HADOOP_CONF_DIR` environment variable or by specifying the `--hadoop-conf` argument to `pydoop submit`.  Note that this is standard pydoop usage, regardless of whether you're using the recently added avro-related features.  Also, Pydoop will always accept full URIs for its input and output paths (as in `hdfs://myhdfshost.mylan:9000/user/me/input_data`).

HTH,

Luca

**Comment by vkuznet** [comment]
Luca, thanks, that clarify things. So, the ```MODULE``` should be a python file name without extension, right? If so, it means that this module should be in a PYTHONPATH.

Another comment, the current top level help
```
pydoop --help
usage: pydoop [-h] [-V] {script,submit} ...

Pydoop command line tool

positional arguments:
  {script,submit}  sub-commands

optional arguments:
  -h, --help       show this help message and exit
  -V, --version    print version number and exit

Supports argparse @confile syntax
```
does not specify that there are sub-command's help section, like ```pydoop submit --help```. I would suggest to write it explicitly in top-level message.

**Comment by vkuznet** [comment]
I'm not sure I understand the INPUT/OUTPUT terminology. The docs said INPUT is the input path to the maps. To which maps? to my input files which MapReduce class will parse and use within a job? And, for the OUTPUT (output path from the reduces). Does it means that the path I specify is where reduce output will be stored. Then the obvious question is which file name will be used for that?

**Comment by vkuznet** [comment]
Yes another question, the example shows that there are two schema files to be used. One which instructs mapper about the input and another which instructs reducer about the output. But example only shows how to tell pydoop tool about output schema. Should we specify input schema as well? For example
```
-D pydoop.mapreduce.avro.value.output.schema="${OUTPUT_SCHEMA}"
-D pydoop.mapreduce.avro.value.input.schema="${INPUT_SCHEMA}"
```

And, if my files are compressed on Hadoop, will MR job works with them or do I need to specify compress option (which one) elsewhere?

**Comment by vkuznet** [comment]
Could you please also clarify the definition of mrv2. The example said that it will work only with Hadoop 2. What exactly does it means? Scanning through code I found that pydoop has helper function
```
    def has_mrv2(self):
        return \
            self.main >= (2, 0, 0) and self.is_yarn() and \
                (not self.is_cloudera() or (self.is_cloudera() and self.dist_version >= (5, 0, 0)))
```

so, if I read it correctly we need to have Hadoop version 2.x.x and above. We also need to have yarn, it should come not from cloudera distribution or from cloudera but with version 5.x.x and above. Am I right?

I use Scientific Linux distribution (compatible with RH) and I installed hadoop from cloudera. My rpm is:
```
hadoop-2.6.0+cdh5.4.2+567-1.cdh5.4.2.p0.4.el6.x86_64
```
so what I'm missing is probably yarn. Ahh, yes, I forgot to say that I got this message when tried to submit my MR job:
```
Avro mode is currently supported only for MRv2
```

Could you please write in plain English the Hadoop requirements to work with Avro.

**Comment by vkuznet** [comment]
The last issue I resolved on my own, my node has no configuration for mapred-site.xml which I updated manually and set yarn as mapreduce.framework.name, etc.

**Comment by vkuznet** [comment]
I figure it out that INPUT corresponds either to concrete file name on HDFS or directory over there. The OUTPUT is an output directory on HDFS where reducer will write its output. I don't need the INPUT_SCHEMA since it is responsibility of my custom Reader class to know how to parse input data. And this eliminates the question about OUTPUT_SCHEMA too. So far, I resolved my misunderstanding of terminology and parameters and closing the ticket.
